### ISSUE TITLE
Length 0 checks should come after checking for mismatched sizes

### ISSUE THREAD

**Issue by syclik** [stan-dev/math#336]
#### Summary:

Bug in checking for length 0 vectorized arguments.
#### Description:

This looks like a bug in the vectorized prob functions (possibly just the cdf / ccdfs). It'll looks something like:

```
// Size checks
if ( !( stan::length(y) && stan::length(mu) && stan::length(sigma) ) )
  return 0.0;
```

This will not check that the dimensions are actually ok. It'll return 0 if any are length 0.

Bob writes in this [pull request comment](https://github.com/stan-dev/math/pull/333#issuecomment-237293394):

> I'd also rather remove all those redundant spaces.  I also
> prefer tighter negations for less nesting and to make it clear
> that if any of the lengths is zero, the result should be zero.
> So I'd write that as:
> 
>   if (!length(y) || !length(mu) || !length(sigma)) return 0;
> 
> And that test should go after the mismatched size tests, which
> would indicate a bug, right?
#### Additional Information:

Provide any additional information here.
#### Current Version:

v2.11.0

**Comment by bob-carpenter** [comment]
I also think these are clearer as 

```
if (!stan::length(y) || !stan::length(mu) || !stan::length(sigma))
  return 0;
```

because the nesting is less deep and it's clearer that if any of them are length zero, we return `0` (don't need that `0.0`).

We should also make sure not to allocate a variable if we're going to just return constant 0 because it can cause autodiff overhead.

**Comment by syclik** [comment]
This is easy, but it'll need to wait until #320 is finished or else we'll end up with lots of merge conflicts.

**Comment by rok-cesnovar** [comment]
Closed via #1207
